client/boot/common.js

```diff
@@ -409,6 +415,10 @@ const boot = async ( currentUser, registerRoutes ) => {
 	utils();
 	await loadPersistedState();
 	const queryClient = await createQueryClient( currentUser?.ID );
+
+	const dehydratedReactQueryState = getDehydratedReactQueryState();
+	hydrate( queryClient, dehydratedReactQueryState );
```

Comment:
Should we avoid hydrating if `dehydratedReactQueryState` is null?

Reply:
`hydrate` will immediately return if the `dehydratedState` is `null`, so an extra check shouldn't be needed.

But I would suggest:
1) moving the server state dehydration code to the same module as `createQueryClient`
2) extracting the persistence code out of `createQueryClient` to a separate function

Then we would call:
```js
const queryClient = createQueryClient(); // it's sync!

await hydrateBrowserState( currentUser?.ID );
hydrateServerState( queryClient, dehydratedState );
```

With these changes you can also avoid a server-side `await` in:
```js
context.queryClient = createQueryClient();
```
because the branch that's async, the persistence code, never runs on the server.